### PR TITLE
Allow Template loading to account for different exception types; add template related tests

### LINKED ISSUE #244: Give more descriptive feedback for loadtemplate command
`loadtemplate FILEPATH` should tell the user why the command has failed, i.e. file not found / invalid format, rather than just a generic error message.

### PR DESCRIPTION
**Pull request by marvintxd**
Resolves #244.

### CODE REVIEW

File: src/test/java/seedu/address/model/template/TemplateTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import seedu.address.model.util.EntryBuilder;
+import seedu.address.testutil.Assert;
+
+public class TemplateTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void addSection_invalidFormat_throwsInvalidTemplateFileException() throws Exception {
```
Comment (jhengy) [inline src/test/java/seedu/address/model/template/TemplateTest.java]: I guess the test method name should be updated?

Comment (jhengy) [reply src/test/java/seedu/address/model/template/TemplateTest.java]: From what I see, this method includes both positive tests and negative tests for Template#addSection()

File: src/main/java/seedu/address/model/template/Template.java [hunk 2]
```diff
@@ -134,7 +147,6 @@ public boolean equals(Object obj) {
 
         // state check
         Template other = (Template) obj;
-        return sections.equals(other.sections)
```
Comment (jhengy) [inline src/main/java/seedu/address/model/template/Template.java]: Just wondering what is the reason to exclude state check for `sections`

File: src/main/java/seedu/address/logic/commands/LoadTemplateCommand.java [hunk 3]
```diff
@@ -28,10 +29,16 @@
         + "Example: " + COMMAND_WORD + " "
         + "template1.txt";
     public static final String MESSAGE_SUCCESS = "Successful load from %1$s.";
-    public static final String MESSAGE_NOT_FOUND = "Loading from %1$s failed.";
+    public static final String MESSAGE_FILE_NOT_FOUND = "File %1$s not found.";
+    public static final String MESSAGE_INVALID_FILE_FORMAT = "File %1$s has invalid format.\n"
+            + "Specified file should consist only of lines of the format: "
+            + "CATEGORY_TILE:~CATEGORY_TAG:[TAG_GROUP]... "
+            + "with no extra newlines or spaces, where each TAG_GROUP is of the format: "
+            + "TAG[&TAG]...";
```
Comment (anubh-v) [inline src/main/java/seedu/address/logic/commands/LoadTemplateCommand.java]: Minor comment: is there a need to add more new lines in the message? Just wondering how the error will look on the result display

File: src/main/java/seedu/address/logic/commands/LoadTemplateCommand.java [hunk 4]
```diff
@@ -28,10 +29,16 @@
...
 
     private final Path filepath;
     private boolean isSuccessful;
+    private Exception exception;
 
     //TESTING
```
Comment (anubh-v) [inline src/main/java/seedu/address/logic/commands/LoadTemplateCommand.java]: What is this comment referring to?

**Comment by anubh-v** [comment]
Not related to this pr: will you be adding tests for `TxtTemplateStorage`?

You can add test templates in the `test/data` folder and use the `equals` method to check if the loaded template equals the expected template.

**Pull request merged**